frontend/src/scene/main-view/CourseModal.js

```diff
@@ -191,14 +195,32 @@ const MainModal = ({
                     <ErrorMessageTag color={errorDetail.colorCode}>
                         {errorDetail.longMessage}
                     </ErrorMessageTag>
-                    <Button
-                        alternative
-                        bold
-                        onClick={onConfirm}
-                        disabled={errorDetail.longMessage}
-                    >
-                        Varaa
-                    </Button>
+                    {!course.reasons[0] ||
```

Comment:
This is still error prone. you need to add check for ´course.reasons´ instead of trying to access the first element from an undefined/null attribute.